resources/assets/js/app/routes.js

```diff
@@ -9,18 +9,35 @@ import NotFound from './layouts/NotFound';
 import RegisterForm from '../features/user/layouts/RegisterForm';
 import RegisterSuccess from '../features/user/layouts/RegisterSuccess';
 import RegisterVerify from '../features/user/layouts/RegisterVerify';
+import LoginForm from '../features/user/layouts/Login/LoginForm';
+import Logout from '../features/user/layouts/Login/Logout';
 
 export default (
     <Route path="/" component={ App }>
         <IndexRoute component={ Home } />
 
-        <Route path="vehicles" component={ Vehicles } />
+        <Route path="vehicles" component={ Vehicles } onEnter={ requireAuth }>
+            // as example to restrict path for unauthenticated users
+            <Route path="vehicles/create" component={ Vehicles } />
+        </Route>
         <Route path="vehicles/:id" component={ VehicleDetails } />
 
         <Route path="registration" component={ RegisterForm } />
         <Route path="registration/success" component={ RegisterSuccess } />
         <Route path="verification" component={ RegisterVerify } />
 
+        <Route path="login" component={ LoginForm } />
+        <Route path="logout" component={ Logout } />
+
         <Route path="*" component={ NotFound } />
     </Route>
 );
+
+function requireAuth(nextState, replace) {
+    if (!sessionStorage.jwt) {
```

Comment:
all work with session should be in AuthManager/AuthService, it should know how to verify if session active or not